agent/api/errors/errors.go

```diff
@@ -95,3 +95,18 @@ func (err *DockerClientConfigError) Error() string { return err.Msg }
 
 // ErrorName returns the name of the error
 func (err *DockerClientConfigError) ErrorName() string { return "DockerClientConfigError" }
+
+// ResourceInitializeError is a task error for which a required resource cannot
+// be initialized
+type ResourceInitializeError struct {
+	TaskArn string
```

Comment:
minor: can this be just `taskARN`? You can have a `New()` method to initialize it instead.